variant/tokenizers/protein_substitution.py

```diff
@@ -0,0 +1,146 @@
+                return None
+
+            ret_psub = True
+        else:
+            # Translation initiation codon: no protein
+            if self.psub['amino_acid'] == '' and self.psub['position'] == 0 \
+                    and self.psub['new_amino_acid'] == '':
+                ret_psub = True
+            # splicing
+            elif not self.psub['amino_acid'] and not self.psub['position'] \
+                    and self.psub['new_amino_acid'] == '?':
+                ret_psub = True
+        if ret_psub:
+            return ProteinSubstitutionToken(
+                token=input_string,
+                input_string=input_string,
+                match_type=TokenMatchType.UNSPECIFIED.value,
+                ref_protein=self.psub['amino_acid'],
+                alt_protein=self.psub['new_amino_acid'],
+                position=self.psub['position']
+            )
+
+        return None
+
+    def _get_psub(self, psub_parts):
+        """Return Protein Substitution tokens.
+
+        :param list psub_parts: The split input string
+        """
+        psub_parts_len = len(psub_parts)
+        if psub_parts_len == 3:
+            if 'p.' in psub_parts[0]:
+                psub_parts[0] = psub_parts[0].split('p.')[-1]
+            else:
+                if not psub_parts[0] and psub_parts[1] and not psub_parts[2]:
+                    return
+
+            if '(' in psub_parts[0] and ')' in psub_parts[2]:
+                psub_parts[0] = psub_parts[0].split('(')[-1]
+                psub_parts[2] = psub_parts[2].split(')')[0]
+
+            self._set_psub(psub_parts[0], psub_parts[1], psub_parts[2])
+
+            if '=/' in self.psub['new_amino_acid']:
+                # mosaic
+                self.psub['new_amino_acid'] = \
+                    self.psub['new_amino_acid'].split('=/')[-1]
+        elif psub_parts_len == 2:
+            psub_parts[0] = psub_parts[0].split(':')[-1]
+            if 'Ter' in psub_parts[0]:
+                check_nonsense = f"({psub_parts[0].replace('Ter', '*')})"
+                if check_nonsense == psub_parts[1]:
+                    psub_parts = \
+                        self.splitter.split(psub_parts[0].split('p.')[-1])
+                    self._get_psub(psub_parts)
+        elif psub_parts_len == 1:
+            if 'p.' in psub_parts[0]:
+                psub_parts[0] = psub_parts[0].split('p.')[1]
+                if psub_parts[0] == '?':
+                    self.psub['new_amino_acid'] = psub_parts[0]
+
+    def _set_psub(self, amino_acid, position, new_amino_acid):
+        """Initialize protein substitution.
+
+        :param str amino_acid: Reference amino acid
+        :param str position: The position of the amino acid substituted
+        :param str new_amino_acid: The new amino_acid
+        """
+        self.psub['amino_acid'] = amino_acid
+        self.psub['position'] = int(position)
+        self.psub['new_amino_acid'] = new_amino_acid
+
+    def _is_valid_amino_acid(self, **kwargs):
```

Comment:
does this need to put the amino acid key behind a **kwargs reference? it seems like the only place it gets called has the correct argument (but I might've missed something)